src/openfermion/ops/_binary_operator.py

```diff
@@ -0,0 +1,494 @@
+# symbolic binary class for decoder definitions (arxiv 1712.07067)
+
+import copy
+import numpy as np
+
+
+class SymbolicBinaryError(Exception):
+    pass
+
+
+class SymbolicBinary(object):
+    """
+        SymbolicBinaries are non-linear binary functions that are expressed analytically, they are
+        components of decoders that are used to rotate terms in qubit basis to fermionic basis. 
+        
+        A SymbolicBinary represents a sum of terms. Each term can consist of two types of binary variables:
+        (n,'W') where n stands for the qubit index, or (1,'1') for constant 1.
```

Comment:
Can you add a little more context to the description about when this class would be used and for what. Also, I think it is a good idea to put the arXiv paper reference in the doc strings for the class so that people looking through the online documentation can find the reference.